src/group_theory/specific_groups/alternating.lean

```diff
@@ -0,0 +1,128 @@
+/-
+Copyright (c) 2021 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Aaron Anderson
+-/
+
+import group_theory.perm.cycle_type
+
+/-!
+# Alternating Groups
+
+The alternating group on a finite type `α` is the subgroup of the permutation group `perm α`
+consisting of the even permutations.
+
+## Main definitions
+
+* `alternating_group α` is the alternating group on `α`, defined as a `subgroup (perm α)`.
+
+## Main results
+* `two_mul_card_alternating_group` shows that the alternating group is half as large as
+  the permutation group it is a subgroup of.
+
+* `closure_three_cycles_eq_alternating` shows that the alternating group is
+  generated by three-cycles.
+
+## Tags
+alternating group permutation
+
+
+## TODO
+* Show that `alternating_group α` is simple if and only if `fintype.card α ≠ 4`.
+
+-/
+
+open equiv equiv.perm subgroup fintype
+variables (α : Type*) [fintype α] [decidable_eq α]
+
+/-- The alternating group on a finite type, realized as a subgroup of `equiv.perm`.
+  For $A_n$, use `alternating_group (fin n)`. -/
+@[derive fintype] def alternating_group : subgroup (perm α) :=
+sign.ker
+
+instance [subsingleton α] : unique (alternating_group α) :=
+⟨⟨1⟩, λ ⟨p, hp⟩, subtype.eq (subsingleton.elim p _)⟩
```

Comment:
Hmm.. Does `infer_instance` find this through `subgroup.unique`?

Reply:
`subgroup.unique` shows that there is only one subgroup, not that a subgroup itself has a unique element.
I bet there is a better API that could be used to provide this instance, but I don't think it exists yet.

Reply:
I'm sure I already went through this thought process last time, only managed to tie it off without your help that time. Sorry for the noise!